internal/pkg/agent/application/upgrade/errors/disk_space.go

```diff
@@ -0,0 +1,30 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package errors
+
+import (
+	"errors"
+)
+
+const insufficientDiskSpaceErrorStr = "insufficient disk space"
+
+var ErrInsufficientDiskSpace = &InsufficientDiskSpaceError{Err: errors.New(insufficientDiskSpaceErrorStr)}
```

Comment:
Could you get away with just the following?
```suggestion
var ErrInsufficientDiskSpace = errors.New(insufficientDiskSpaceErrorStr)
```

If so, you wouldn't need the type below.